eth/backend.go

```diff
@@ -131,12 +133,28 @@ func New(stack *node.Node, config *ethconfig.Config) (*Ethereum, error) {
 	ethashConfig := config.Ethash
 	ethashConfig.NotifyFull = config.Miner.NotifyFull
 
+	var chainDb ethdb.Database
+	var err error
+	var chainConfig *params.ChainConfig
+	var genesisHash common.Hash
+	var genesisErr error
 	// Assemble the Ethereum object
-	chainDb, err := stack.OpenDatabaseWithFreezer("chaindata", config.DatabaseCache, config.DatabaseHandles, config.DatabaseFreezer, "eth/db/chaindata/", false)
+	if readOnly {
+		chainDb, err = stack.OpenDatabase("chaindata", config.DatabaseCache, config.DatabaseHandles, "eth/db/chaindata/", true)
+		if config.Genesis != nil {
+			chainConfig = config.Genesis.Config
+		} else {
+			chainConfig = core.DefaultGenesisBlock().Config
+		}
+		genesisHash = common.Hash{}
+		genesisErr = nil
+	} else {
+		chainDb, err = stack.OpenDatabaseWithFreezer("chaindata", config.DatabaseCache, config.DatabaseHandles, config.DatabaseFreezer, "eth/db/chaindata/", false)
```

Comment:
If the database cannot be opened read-only with a freezer, why does `OpenDatabaesWithFreezer` have a `readonly` arg? Please comment this for the reader.